cmd/snap-preseed/main.go

```diff
@@ -103,17 +111,36 @@ func run(parser *flags.Parser, args []string) error {
 		return resetPreseededChroot(chrootDir)
 	}
 
-	if err := checkChroot(chrootDir); err != nil {
-		return err
-	}
-
-	targetSnapd, cleanup, err := prepareChroot(chrootDir)
-	if err != nil {
-		return err
+	var cleanup func()
+	if opts.Core20 {
+		var popts *PreseedOpts
+		popts, cleanup, err = prepareCore20Chroot(chrootDir)
+		if err != nil {
+			return err
+		}
+
+		err = runUC20PreseedMode(popts)
+	} else {
+		if err := checkChroot(chrootDir); err != nil {
+			return err
+		}
+
+		var targetSnapd *targetSnapdInfo
+
+		// XXX: if prepareClassicChroot & runPreseedMode were refactored to
+		// use "chroot" inside runPreseedMode (and not syscall.Chroot at the
+		// beginning of prepareClassicChroot, then we could have a single
+		// runPreseedMode/runUC20PreseedMode function that handles both classic
+		// and core20).
```

Comment:
There's something wrong with the parentheses in this comment (or there is an "if" too many at the beginning?).